src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    internal static class OrderingHelper
+    {
+        private enum MoveAction { Up=0, Down=1 }
+
+        private static void ProcessProjectTree(Queue<IProjectTree> treeQueue, HashSet<string> lookup, UnconfiguredProject unconfiguredProject, IProjectTree projectTree)
```

Comment:
I don't like that this method modifies both `treeQueue` and `lookup` but returns `void`.  Perhaps return a tuple so its clear what the inputs and outputs are?